flask/app/analyses.py

```diff
@@ -23,60 +27,114 @@
 
 @analyses_blueprint.route("/api/analyses", methods=["GET"], endpoint="analyses_list")
 @login_required
-def list_analyses():
+@paged
+def list_analyses(page: int, limit: int) -> Response:
+    order_by = request.args.get("order_by", type=str)
+    allowed_columns = [
+        "updated",
+        "result_path",
+        "notes",
+        "analysis_state",
+        "assignee",
+        "requester",
+    ]
+    assignee_user = aliased(models.User)
+    requester_user = aliased(models.User)
+    if order_by is None:
+        order = None  # system default, likely analysis_id
+    elif order_by == "assignee":
+        order = assignee_user.username
+    elif order_by == "requester":
+        order = requester_user.username
+    elif order_by in allowed_columns:
+        # Since this is an elif clause, we know special cases are already handled above.
+        # order_by has been restricted to a known list, so we can safely use getattr
+        order = getattr(models.Analysis, order_by)
+    else:
+        abort(400, description=f"order_by must be one of {allowed_columns}")
+
+    if order:
+        order_dir = request.args.get("order_dir", type=str)
+        if order_dir == "desc":
+            order = order.desc()
+        elif order_dir == "asc":
+            order = order.asc()
+        else:
+            abort(400, description="order_dir must be either 'asc' or 'desc'")
+
+    filters = []
+    assignee = request.args.get("assignee", type=str)
+    if assignee:
+        filters.append(func.instr(assignee_user.username, assignee))
+    requester = request.args.get("requester", type=str)
+    if requester:
+        filters.append(func.instr(requester_user.username, requester))
+    notes = request.args.get("notes", type=str)
+    if notes:
+        filters.append(func.instr(models.Analysis.notes, notes))
+    result_path = request.args.get("result_path", type=str)
+    if result_path:
+        filters.append(func.instr(models.Analysis.result_path, result_path))
+    updated = request.args.get("updated", type=str)
+    if updated:
+        filters.append(filter_updated_or_abort(models.Analysis.updated, updated))
+    analysis_state = request.args.get("analysis_state", type=str)
+    if analysis_state:
+        filters.append(
+            filter_in_enum_or_abort(
+                models.Analysis.analysis_state,
+                models.AnalysisState,
+                analysis_state,
+            )
+        )
+
     if app.config.get("LOGIN_DISABLED") or current_user.is_admin:
         user_id = request.args.get("user")
     else:
         user_id = current_user.user_id
-
-    since_date = request.args.get("since", default="0001-01-01T00:00:00-04:00")
-    try:
-        since_date = datetime.fromisoformat(since_date)
-    except:
-        abort(400, description="Malformed query date")
```

Comment:
The `since` parameter is needed for the notifications feature. It's only ever invoked by itself, without any other query parameters.

Reply:
I expect that it can be updated to use `updated=after,iso-datetime`

Reply:
Ah that makes sense.